flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/reader/SnapshotSplitReader.java

```diff
@@ -288,7 +291,10 @@ public Iterator<SourceRecord> pollSplitRecords() throws InterruptedException {
             normalizedRecords.add(lowWatermark);
             normalizedRecords.addAll(formatMessageTimestamp(snapshotRecords.values()));
             normalizedRecords.add(highWatermark);
-            return normalizedRecords.iterator();
+
+            final Set<SourceRecords> sourceRecordsSet = new HashSet<>();
```

Comment:
I see that we use `List` in `BinlogSplitReader`. So we better use `List` here as well.